### PR TITLE
- fix list_distance & list_cosine_similarity execute fail when list i…

### LINKED ISSUE #12960: list_cosine_similarity([], []) gives -1
### What happens?

I'm not positive what the expected behavior should be, it's not particularly well-defined, but I think NULL would be a lot better than -1. -1 means "pointing in opposite directions" which isn't quite right.

### To Reproduce

```sql
SELECT list_cosine_similarity([],[]);`
```
results in -1

### OS:

https://shell.duckdb.org/

### DuckDB Version:

v1.0.0

### DuckDB Client:

https://shell.duckdb.org/

### Full Name:

Nick Crews

### Affiliation:

Ship Creek Group

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

### PR DESCRIPTION
**Pull request by flashmouse**
According to #8750, there are still some list_xx(list_cosine_similarity, list_distance) functions have the same error like #13045 not fixed in #13076.
And this pr also try to fix #12960

### CODE REVIEW

File: src/core_functions/scalar/list/list_cosine_similarity.cpp [hunk 1]
```diff
@@ -41,6 +45,10 @@ static void ListCosineSimilarity(DataChunk &args, ExpressionState &, Vector &res
 
 		    auto dimensions = left.length;
 
+		    if (dimensions == 0) {
+			    return NullValue<NUMERIC_TYPE>();
```
Comment (Mytherin) [inline src/core_functions/scalar/list/list_cosine_similarity.cpp:49]: This now returns `NaN`. Can we rework this to throw an error instead? (e.g. `InvalidInputException("The cosine similarity for empty vectors is not defined")`).

**Comment by Mytherin** [comment]
Thanks! LGTM - one comment

File: src/core_functions/scalar/list/list_cosine_similarity.cpp (hunk 1 above)
Comment (flashmouse) [reply src/core_functions/scalar/list/list_cosine_similarity.cpp]: updated, plz check again

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**